sparkle/lib/target/risc-v/encoder/branch.cpp

```
@@ -0,0 +1,163 @@
+        if (offset < -4096 || offset > 4094)
+            /* if out of range, set the offset to maximum */
+            offset = (offset < 0) ? -4096 : 4094;
+        
+        /* extract bits to get the B-type immediate */
+        int imm12 = (offset >> 1) & 0xFFF;
+        int imm_12 = (imm12 >> 11) & 0x1;
+        int imm_11 = (imm12 >> 10) & 0x1;
+        int imm_10_5 = (imm12 >> 4) & 0x3F;
+        int imm_4_1 = imm12 & 0xF;
+        
+        return (imm_12 << 31) |     /* imm[12] */
```

Cppcheck (2.10-2)

[error] Signed integer overflow for expression 'imm_12<<31'.

(integerOverflow)